commands/acr-build.ts

```diff
@@ -0,0 +1,99 @@
+import { ContainerRegistryManagementClient } from 'azure-arm-containerregistry';
+import { QuickBuildRequest } from "azure-arm-containerregistry/lib/models";
+import { Registry } from 'azure-arm-containerregistry/lib/models';
+import { ResourceManagementClient } from 'azure-arm-resource';
+import { BlobService, createBlobServiceWithSas } from "azure-storage";
+import * as vscode from "vscode";
+import { ImageNode } from "../explorer/models/imageNode";
+import { AzureUtilityManager } from "../utils/azureUtilityManager";
+import { acquireResourceGroup, acquireSubscription, quickPickACRRegistry } from './utils/quick-pick-azure';
+let tar = require('tar');
+let fs = require('fs');
+let os = require('os');
+let url = require('url');
+
+export async function queueBuild(context?: ImageNode): Promise<void> {
+
+    console.log("Obtaining Subscription and Client");
+    let subscription = await acquireSubscription();
+    let client = AzureUtilityManager.getInstance().getContainerRegistryManagementClient(subscription);
+    const resourceGroupClient = new ResourceManagementClient(AzureUtilityManager.getInstance().getCredentialByTenantId(subscription.tenantId), subscription.subscriptionId);
+
+    let resourceGroup = await acquireResourceGroup(subscription, resourceGroupClient);
+    let resourceGroupName = resourceGroup.name;
+
+    let registry: Registry = await quickPickACRRegistry(subscription, resourceGroupName);
+    let registryName = registry.name;
+
+    console.log("Setting up temp file with 'sourceArchive.tar.gz' ");
+    let tarFilePath = url.resolve(os.tmpdir(), 'sourceArchive.tar.gz');
+
+    console.log("Uploading Source Code");
+    let sourceLocation: string = vscode.workspace.rootPath;
```

Comment:
>sourceLocation [](start = 8, length = 14)

The sourceLocation should be the dockerfile folder. 
In reality, dockerfile may not be the root folder of the source code, eg,

The uploaded source  folder is: rootPath/src/
The dockerfile is: rootPath/src/folder1/Dockerfile